Lib/test/test_tokenize.py

```diff
@@ -922,12 +934,15 @@ def baz(): pass
 class GenerateTokensTest(TokenizeTest):
     def check_tokenize(self, s, expected):
         # Format the tokens in s in a table format.
-        # The ENDMARKER is omitted.
+        # The ENDMARKER and final NEWLINE are omitted.
         result = []
         f = StringIO(s)
+        num_lines = len(s.splitlines())
         for type, token, start, end, line in generate_tokens(f.readline):
             if type == ENDMARKER:
                 break
+            if s[-1] not in '\r\n' and type == NEWLINE and end[0] == num_lines:
+                continue
```

Comment:
The `check_tokenize()` logic is repeated exactly, but it is now becoming rather involved. This should be a common function used by both classes.